src/java.base/share/classes/jdk/internal/util/HexDigits.java

```diff
@@ -114,24 +114,17 @@ public static short digitPair(int i, boolean ucase) {
     }
 
     /**
-     * Return a little-endian packed integer for the 4 ASCII bytes for an input unsigned 2-byte integer.
-     * {@code b0} is the most significant byte and {@code b1} is the least significant byte.
-     * The integer is passed byte-wise to allow reordering of execution.
+     * Insert the int into the buffer as 4 hexadecimal digits
+     * @param buffer byte buffer to copy into
+     * @param off insert point
+     * @param value to convert
      */
-    public static int packDigits(int b0, int b1) {
-        return DIGITS[b0 & 0xff] | (DIGITS[b1 & 0xff] << 16);
-    }
-
-    /**
-     * Return a little-endian packed long for the 8 ASCII bytes for an input unsigned 4-byte integer.
-     * {@code b0} is the most significant byte and {@code b3} is the least significant byte.
-     * The integer is passed byte-wise to allow reordering of execution.
-     */
-    public static long packDigits(int b0, int b1, int b2, int b3) {
-        return DIGITS[b0 & 0xff]
-                | (DIGITS[b1 & 0xff] << 16)
-                | (((long) DIGITS[b2 & 0xff]) << 32)
-                | (((long) DIGITS[b3 & 0xff]) << 48);
+    public static void putHex(byte[] buffer, int off, int i) {
+        int v = (DIGITS[i & 0xff] << 16) | DIGITS[(i >> 8) & 0xff];
```

Comment:
Looking at the comments in #16245, it's said that the int must be computed beforehand so it will be a write of a 4-byte instead of 2 write of 2-bytes. Is that understanding correct?

If my understanding is correct, I recommend to leave a comment explaining why we compact the value into an int, like
```
// Prepare an int value so C2 generates a 4-byte write instead of two 2-byte writes
```